spinalcordtoolbox/reports/slice.py

```diff
@@ -294,7 +322,10 @@ def single(self):
         return matrices
 
     def aspect(self):
-        return [self.get_aspect(x) for x in self._images]
+        if len(self._4d_images) == 0:  # For 3D images
+            return [self.get_aspect(x) for x in self._images]
+        else:  # For 4D images
+            return [self.get_aspect(x) for x in self._4d_images]
```

Comment:
e.g. this change would no longer be necessary if you re-used `self._images` for the 4D case.